src/topology/vector_bundle.lean

```diff
@@ -0,0 +1,221 @@
+topology on continuous linear maps between general topological vector spaces). Let
+`vector_bundle_continuous_linear_map 𝕜 F₁ E₁ F₂ E₂ (x : B)` be a type synonym for `E₁ x →L[𝕜] E₂ x`.
+Then one can endow
+`bundle.total_space (vector_bundle_continuous_linear_map 𝕜 F₁ E₁ F₂ E₂)`
+with a topology and a topological vector bundle structure.
+
+Similar constructions can be done for tensor products of topological vector bundles, exterior
+algebras, and so on, where the topology can be defined using a norm on the fiber model if this
+helps.
+-/
+
+noncomputable theory
+
+
+open bundle
+
+variables (𝕜 : Type*) {B : Type*} (F : Type*) (E : B → Type*)
+[normed_field 𝕜] [∀ x, add_comm_group (E x)] [∀ x, semimodule 𝕜 (E x)]
+[topological_space F] [add_comm_group F] [semimodule 𝕜 F]
+[topological_space (total_space E)] [topological_space B]
+
+section
+
+/-- Local trivialization for vector bundles. -/
+@[nolint has_inhabited_instance]
+structure topological_vector_bundle.trivialization extends bundle_trivialization F (proj E) :=
+(linear : ∀ x ∈ base_set, is_linear_map 𝕜 (λ y : (E x), (to_fun y).2))
+
+instance : has_coe_to_fun (topological_vector_bundle.trivialization 𝕜 F E) :=
+⟨λ _, (total_space E → B × F), λ e, e.to_bundle_trivialization⟩
+
+end
+
+variables [∀ x, topological_space (E x)]
+
+/-- The space `total_space E` (for `E : B → Type*` such that each `E x` is a topological vector
+space) has a topological vector space structure with fiber `F` (denoted with
+`topological_vector_bundle 𝕜 F E`) if around every point there is a fiber bundle trivialization
+which is linear in the fibers. -/
+class topological_vector_bundle : Prop :=
+(inducing [] : ∀ (b : B), inducing (λ x : (E b), (id ⟨b, x⟩ : total_space E)))
+(locally_trivial [] : ∀ b : B, ∃ e : topological_vector_bundle.trivialization 𝕜 F E, b ∈ e.base_set)
+
+variable [topological_vector_bundle 𝕜 F E]
+
+namespace topological_vector_bundle
+
+/-- `trivialization_at 𝕜 F E b` is some choice of trivialization of a vector bundle whose base set
+contains a given point `b`. -/
+def trivialization_at : Π b : B, trivialization 𝕜 F E :=
+λ b, classical.some (topological_vector_bundle.locally_trivial 𝕜 F E b)
+
+@[simp, mfld_simps] lemma mem_base_set_trivialization_at (b : B) :
+  b ∈ (trivialization_at 𝕜 F E b).base_set :=
+classical.some_spec (topological_vector_bundle.locally_trivial 𝕜 F E b)
+
+@[simp, mfld_simps] lemma mem_source_trivialization_at (z : total_space E) :
+  z ∈ (trivialization_at 𝕜 F E z.1).source :=
+by { rw bundle_trivialization.mem_source, apply mem_base_set_trivialization_at }
+
+/-- In a topological vector bundle, a trivialization in the fiber (which is a priori only linear)
+is in fact a continuous linear equiv between the fibers and the model fiber. -/
+def trivialization.continuous_linear_equiv_at (e : trivialization 𝕜 F E) (b : B)
+  (hb : b ∈ e.base_set) : continuous_linear_equiv 𝕜 (E b) F :=
```

Comment:
Doesn't this need a `@[simp]` lemma relating it back to the original `e`?

Reply:
You mean something of this kind?
```
@[simp] lemma trivialization.continuous_linear_equiv_at_evaluation (e : trivialization 𝕜 F E)
  (x : total_space E) (hx : x ∈ e.source) :
  e.continuous_linear_equiv_at (proj E x) (e.mem_source.1 hx) x.2 = (e x).2 :=
by { cases x, refl }
```
What do you mean by "in both directions"?

Reply:
No, I wanted something where all the arguments are free variables, not constructed out of more complicated things. Basically so that when this gadget is "fully applied", `simp` will take it apart. So something more like

```
@[simp] trivialization.continuous_linear_equiv_at_apply (e : trivialization 𝕜 F E) (b : B)
  (hb : b ∈ e.base_set) (y) : e.continuous_linear_equiv_at b hb y = (e ⟨b, y⟩).2 := rfl
```

And then "both directions" meant a corresponding lemma for `(e.continuous_linear_equiv_at b hb).symm z`, but that is going to be too gross to be useful, so ignore that suggestion!

Reply:
Ok sure thanks! Your lemma is definitely simpler than mine, but still I tried and it seems it does not directly cover mine (because lean does not automatically recognize `e \<x.1, x.2\>` as `e x`), so I wonder if it might be useful to include my version as well